### PR TITLE
[FR] DUNI-2664 Completely refactored dockerfile

### PR DESCRIPTION
**Pull request by froblesmartin**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
+
+ENV DBUS_SESSION_BUS_ADDRESS=/dev/null \
+    TINI_VERSION=v0.18.0 \
+    CHROME_VERSION=76.0.3809.132 \
+    CHROME_DRIVER_VERSION=76.0.3809.126 \
+    PROTRACTOR_VERSION=5.4.2 \
+    PYTHON=/usr/bin/python2.7
+
+# Sets timezone defined in TZ environmental variable
+ENV TZ=Europe/London
```
Comment (Eusebiotrigo) [inline Dockerfile]: Could it be Europe/Berlin? Or Parametrized by project?

File: Dockerfile [hunk 2]
```diff
@@ -1,33 +1,52 @@
...
-RUN gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
-RUN curl -L get.rvm.io | bash -s stable
-RUN /bin/bash -l -c '. /home/jenkins/.rvm/scripts/rvm && rvm install ${RUBY_VERSION}'
-RUN cat /home/jenkins/.profile >> /home/jenkins/.bash_profile
-RUN mkdir /home/jenkins/bots
-RUN /bin/bash -l -c "gem install bundler"
+
+RUN mkdir /home/jenkins/bots && \
+    touch /home/jenkins/xvfb.log && \
+    touch /home/jenkins/Xauthority
```
Comment (bardyr) [inline Dockerfile]: Why do you have 
` touch /home/jenkins/xvfb.log && \
RUN /bin/bash -c ". ~/.nvm/nvm.sh && \	    touch /home/jenkins/Xauthority`

tests should run headless without xvfb

File: Dockerfile [hunk 3]
```diff
@@ -1,33 +1,52 @@
...
 
-RUN dnf -y update &&  dnf -y install gpg autoconf automake bison bzip2 gcc-c++ glibc-headers glibc-devel libffi-devel libtool libyaml-devel readline-devel sqlite-devel zlib-devel openssl-devel java-1.8.0-openjdk python2 && \
-	dnf clean all --enablerepo='*'
+# Fix to allow installing openjdk-8 (taken from https://github.com/nextcloud/docker/issues/380)
+RUN mkdir -p /usr/share/man/man1mkdir -p /usr/share/man/man1
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    xvfb \
+    wget \
+    openjdk-8-jre-headless \
```
Comment (Eusebiotrigo) [inline Dockerfile]: do we need java?

File: Dockerfile [hunk 4]
```diff
@@ -1,33 +1,52 @@
...
-RUN curl -OSL https://s3-eu-west-1.amazonaws.com/e2y-hybris/chrome/google-chrome-stable-${CHROME_VERSION}-1.x86_64.rpm
-RUN dnf localinstall -y google-chrome-stable-${CHROME_VERSION}-1.x86_64.rpm && \
-	dnf clean all --enablerepo='*'
-RUN sed -in--place "s/\$@/\$@ --disable-gpu --no-sandbox/g" /opt/google/chrome/google-chrome
 RUN useradd jenkins --shell /bin/bash --create-home
 
+RUN bin/bash -c "npm config set python /usr/bin/python2.7 && \
+    npm install -g protractor@^${PROTRACTOR_VERSION} && \
+    webdriver-manager update --versions.chrome=${CHROME_DRIVER_VERSION} && \
+    webdriver-manager update"
```
Comment (bardyr) [inline Dockerfile]: is the update command supposed to be duplicated?

Comment (Eusebiotrigo) [inline Dockerfile]: do we need this twice?

Comment (froblesmartin) [reply Dockerfile]: I think so, first one updates ChromeDriver and second regular stuff, will need @AlvaroMe2y to confirm it

File: Dockerfile (hunk 1 above)
Comment (froblesmartin) [reply Dockerfile]: I configured it to be the same time as Jenkins and AWS instances, I guess that we should keep everything on that time to do not have a huge mess.

File: Dockerfile (hunk 4 above)
Comment (froblesmartin) [reply Dockerfile]: I have tried without it and fails

Comment (froblesmartin) [reply Dockerfile]: Yes, it fails if not. @AlvaroMe2y should know better why :sweat_smile:

File: Dockerfile (hunk 3 above)
Comment (froblesmartin) [reply Dockerfile]: Sadly, yes. Selenium uses it.

File: Dockerfile (hunk 2 above)
Comment (froblesmartin) [reply Dockerfile]: True story. I read something that made me think that Cucumber used xvfb behind the scenes, but it doesn't or at least it doesn't need to have installed manually.

**Pull request merged**